client/signup/steps/plans/index.jsx

```diff
@@ -146,6 +146,67 @@ export class PlansStep extends Component {
 			);
 		}
 
+		if ( isMobile() && 'onboarding' === this.props.flowName ) {
+			return (
+				<ProvideExperimentData name="calypso_mobile_plans_page_with_billing">
```

Comment:
Instead of the [conditional above](https://github.com/Automattic/wp-calypso/pull/58981/files#diff-f4523a33abda8414624dfd8061cffa9cc3b52e263f9a755706831b28b7b55cecR149) and the [final return statement](https://github.com/Automattic/wp-calypso/pull/58981/files#diff-f4523a33abda8414624dfd8061cffa9cc3b52e263f9a755706831b28b7b55cecR195) we can use the `isEligible` check (read more [here](https://github.com/Automattic/wp-calypso/tree/trunk/packages/explat-client-react-helpers#experimentoptions)) for the experiment helper and the functionality will remain unchanged:

```suggestion
				<ProvideExperimentData
					name="calypso_mobile_plans_page_with_billing"
					options={ { isEligible: isMobile() && 'onboarding' === this.props.flowName } }
				>
```